src/vcpkg/vcpkgpaths.cpp

```diff
@@ -672,9 +679,41 @@ namespace vcpkg
                                                        m_pimpl->m_config_dir,
                                                        *this);
 
+            auto make_relative_to_manifest = [&](std::string& overlay_path) -> std::string {
+                auto p = Path(overlay_path);
+                if (p.is_relative())
+                {
+                    auto maybe_manifest_dir = get_manifest_directory();
+                    if (const auto manifest_dir = maybe_manifest_dir.get())
+                    {
+                        Path tmp(*manifest_dir);
+                        auto res = tmp / p;
+                        overlay_path = res.generic_u8string();
+                        return overlay_path;
+                    }
+                }
+                return overlay_path;
```

Comment:
In total (untested):
~~~c++
auto const p = Path(overlay_path);
return p.is_relative() ? std::string(manifest_dir / overlay_path) : overlay_path;
~~~